### PR TITLE
Multisig support

### PR DESCRIPTION
**Pull request by darosior**
This permits the creation of a Liana descriptor containing a multisig in either of the two spending paths. The multisig is currently restricted at creation time to be a CHECKUMULTISIG-like multisig (k-of-n with n<20). See the linked issue for rationale.

Closes https://github.com/wizardsardine/liana/issues/53.

### CODE REVIEW

**Comment by darosior** [comment]
I'll need to open an issue to track the support for more advanced multisigs.

**Comment by darosior** [comment]
I'll need to open an issue to track the support for more advanced multisigs.

**Comment by darosior** [comment]
Ah-ha! Found an issue with this. The way we lift the policy to get the CSV threshold is incorrect.

We should consider having a fuzz target for this soon.

**Comment by darosior** [comment]
Pushed a couple commits that introduce a helper to analyze a partially signed transaction. It makes a few assumptions, but hopefully it can be useful to have a nice presentation in the GUI.

File: src/descriptors.rs [hunk 1]
```diff
@@ -333,12 +434,103 @@ impl PartialEq<descriptor::Descriptor<descriptor::DescriptorPublicKey>> for Inhe
...
+pub struct PathSpendInfo {
+    pub sigs_count: usize,
+    pub threshold: usize,
+}
+
+/// Information about a partial spend of Liana coins
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Ord, PartialOrd, Hash)]
+pub struct PartialSpendInfo {
+    /// Number of signatures present for the primary path
+    primary_path: PathSpendInfo,
```
Comment (edouardparis) [inline src/descriptors.rs:522]: Please make it public or provide some methods to access it
Same for recovery_path

File: src/descriptors.rs [hunk 2]
```diff
@@ -333,12 +434,103 @@ impl PartialEq<descriptor::Descriptor<descriptor::DescriptorPublicKey>> for Inhe
...
+    /// Timelock and path info for the recovery path.
+    pub fn recovery_path(&self) -> (u16, &PathInfo) {
+        (self.recovery_path.0, &self.recovery_path.1)
+    }
+}
+
+/// Number of signatures present and required number of signatures for a given path.
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Ord, PartialOrd, Hash)]
+pub struct PathSpendInfo {
+    pub sigs_count: usize,
```
Comment (edouardparis) [inline src/descriptors.rs:514]: It is more useful for the GUI to provide the (fingerprint,derivation_path) of each signing key. So user can know which one is missing.

**Comment by darosior** [comment]
@edouardparis i addressed your requests. I think you told me about a crash too, do you have a repro?

**Comment by darosior** [comment]
Alright, i pushed an extended (but not quite comprehensive) set of unit test for the `partial_spend_info` method, along with a fix. I think this PR is now in a pretty final state.

**Review by edouardparis — APPROVED** [review]
ACK 3ea85fa9506bc96440096a9d11cceffdbf5eab44

**Pull request merged**